backend/remote-state/azure/arm_client.go

```diff
@@ -0,0 +1,119 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/resources/mgmt/resources"
+	armStorage "github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2016-05-01/storage"
```

Comment:
we'll probably want to use [the Profile](https://github.com/Azure/azure-sdk-for-go/blob/v21.3.0/profiles/2017-03-09/storage/mgmt/storage/models.go) here to ensure compatibility with Azure Stack